watertap/examples/flowsheets/case_studies/groundwater_treatment/tests/test_groundwater_treatment.py

```diff
@@ -0,0 +1,162 @@
+
+from watertap.examples.flowsheets.case_studies.groundwater_treatment.groundwater_treatment import (
+    build,
+    set_operating_conditions,
+    initialize_system,
+    solve,
+    add_costing,
+    display_costing,
+    display_results,
+)
+
+
+class TestGroundwaterTreatmentFlowsheet:
+    @pytest.fixture(scope="class")
+    def system_frame(self):
+        m = build()
+        return m
+
+    @pytest.mark.unit
+    def test_build(self, system_frame):
+        m = system_frame
+        assert_degrees_of_freedom(m, 19)
+        assert_units_consistent(m)
+
+    @pytest.mark.component
+    def test_set_operating_conditions(self, system_frame):
+        m = system_frame
+        set_operating_conditions(m)
+
+        # test feed
+        assert pytest.approx(0.011574, rel=1e-3) == value(
+            m.fs.feed.flow_mass_comp[0, "H2O"]
+        )
+
+        assert pytest.approx(4.6293e-10, rel=1e-3) == value(
+            m.fs.feed.flow_mass_comp[0, "arsenic"]
+        )
+
+        assert pytest.approx(6.94444e-10, rel=1e-3) == value(
+            m.fs.feed.flow_mass_comp[0, "uranium"]
+        )
+
+        assert pytest.approx(1.15741e-7, rel=1e-3) == value(
+            m.fs.feed.flow_mass_comp[0, "nitrate"]
+        )
+
+        assert pytest.approx(1.15741e-9, rel=1e-3) == value(
+            m.fs.feed.flow_mass_comp[0, "phosphates"]
+        )
+
+        assert pytest.approx(5.787037e-9, rel=1e-3) == value(
+            m.fs.feed.flow_mass_comp[0, "iron"]
+        )
+
+        assert pytest.approx(1.15741e-9, rel=1e-3) == value(
+            m.fs.feed.flow_mass_comp[0, "filtration_media"]
+        )
+
+        # test pump head
+        assert pytest.approx(30, rel=1e-3) == value(m.fs.pump.lift_height)
+
+    @pytest.mark.component
+    def test_initialize(self, system_frame):
+        m = system_frame
+        initialize_system(m)
+
+    @pytest.mark.component
+    def test_solve(self, system_frame):
+        m = system_frame
+        results = solve(m)
+        assert_optimal_termination(results)
+
+        # check products - treated water
+        assert pytest.approx(0.011574, rel=1e-3) == value(
+            m.fs.filtered_water.flow_mass_comp[0, "H2O"]
+        )
+
+        assert pytest.approx(4.62963e-11, rel=1e-3) == value(
+            m.fs.filtered_water.flow_mass_comp[0, "arsenic"]
+        )
+
+        assert pytest.approx(6.94444e-11, rel=1e-3) == value(
+            m.fs.filtered_water.flow_mass_comp[0, "uranium"]
+        )
+
+        assert pytest.approx(2.893519e-8, rel=1e-3) == value(
+            m.fs.filtered_water.flow_mass_comp[0, "nitrate"]
+        )
+
+        assert pytest.approx(1.157407e-10, rel=1e-3) == value(
+            m.fs.filtered_water.flow_mass_comp[0, "phosphates"]
+        )
+
+        assert value(m.fs.filtered_water.flow_mass_comp[0, "iron"]) < 1e-12
+
+        assert value(m.fs.filtered_water.flow_mass_comp[0, "filtration_media"]) < 1e-12
```

Comment:
Because the amount of filtration media in the filtered water stream should essentially be 0. Notice in the yaml file that the removal fraction of `filtration_media` is 1.

Reply:
I figured that much-- just nitpicking here.